src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/tests/latest/test_resource.py

```diff
@@ -791,14 +803,25 @@ def test_resource_policyset_default(self, resource_group):
     @ResourceGroupPreparer(name_prefix='cli_test_policyset_management_group')
     @AllowLargeResponse()
     def test_resource_policyset_management_group(self, resource_group):
-        self.resource_policyset_operations(resource_group, 'AzGovTest8')
+        management_group_name = self.create_random_name('cli-test-mgmt-group', 30)
+        self.cmd('account management-group create -n ' + management_group_name)
+        try:
+            self.resource_policyset_operations(resource_group, management_group_name)
+        finally:
+            self.cmd('account management-group delete -n ' + management_group_name)
 
+    @record_only()
     @ResourceGroupPreparer(name_prefix='cli_test_policyset_subscription_id')
     @AllowLargeResponse()
     def test_resource_policyset_subscription_id(self, resource_group):
-        self.resource_policyset_operations(resource_group, None, 'e8a0d3c2-c26a-4363-ba6b-f56ac74c5ae0')
+        if not (self.in_recording or self.is_live):
```

Comment:
when is running a test not live or in a recording?

Reply:
it is running under playback, where all subscriptions should be `00000....`

For context, this whole test class has never passed no matter under playback or live.

Reply:
I see, I confused in a recording with playback. 
Just a small detail but, `in_recording` covers the condition of `is_live`: [info](https://github.com/Azure/azure-python-devtools/blob/master/src/azure_devtools/scenario_tests/base.py#L123)

Reply:
Thank you. I have simplified the code per your comment and also added comments for why doing this.